torch/_dynamo/variables/torch.py

```diff
@@ -676,11 +677,26 @@ def fn_with_prim_types(x):
                 and self.value in tensortype_to_dtype
                 and len(args) == 1
                 and isinstance(args[0], ListVariable)
-                and args[0].is_python_constant()
             ):
-                for x in args[0].items:
-                    if isinstance(x.value, np.generic):
-                        x.value = x.value.item()
+                any_ndarray = any(
+                    isinstance(x, variables.NumpyNdarrayVariable) for x in args[0].items
+                )
+                # we cannot torch.stack non-tensor like
+                all_tensor_like = all(
+                    isinstance(x, variables.TensorVariable) for x in args[0].items
+                )
+                if any_ndarray and all_tensor_like:
```

Comment:
We can avoid putting anything in the graph if the list has just one tensor/array (as in the test that this PR is fixing)

Reply:
I think this is not true. With 1 array, you still need to stack to get a tensor, as it doesn't accept a list of tensors.